Test/Graph/unit-graph.cc

```diff
@@ -30,9 +33,48 @@ TEST_CASE("graphs have consistent number of sites", "[graph]") {
 
     SECTION("Graph test (" + std::to_string(i) + ") on " + name) {
 
-      netket::Graph graph(input_tests[i]);
+      Graph graph(input_tests[i]);
 
       REQUIRE(graph.Nsites() > 0);
     }
   }
 }
+
+TEST_CASE("Breadth-first search", "[graph]") {
+  const auto input_tests = GetGraphInputs();
+
+  for(const auto input : input_tests) {
+    const auto data = input.dump();
+    SECTION("each node is visited at most once on " + data) {
+      Graph graph(input);
+      for(int start = 0; start < graph.Nsites(); ++start) {
+        std::unordered_set<int> visited;
+        int ncall = 0;
+        graph.BreadthFirstSearch(start, [&](int v, int depth) {
+          INFO("ncall: " << ncall << ", start: " << start << ", v: " << v << ", depth: " << depth);
+          REQUIRE(visited.count(v) == 0);
+          visited.insert(v);
+          ncall++;
+        });
+      }
+    }
+  }
+
+  SECTION("each node is visited at least once on connected graph") {
```

Comment:
Now that you mention this: we don't have a function to check if a graph is connected or not! That (simple) function would also make this test look nicer in terms of input

Reply:
True. I can add that in this PR in a moment.